app/models/katello/host/subscription_facet.rb

```diff
@@ -43,6 +45,9 @@ def import_database_attributes(consumer_params = candlepin_consumer.consumer_att
         self.registered_at = consumer_params['created'] unless consumer_params['created'].blank?
         self.last_checkin = consumer_params['lastCheckin'] unless consumer_params['lastCheckin'].blank?
         self.update_installed_products(consumer_params['installedProducts']) if consumer_params.key?('installedProducts')
+        self.role = consumer_params['role'] unless consumer_params['role'].blank?
+        self.usage = consumer_params['usage'] unless consumer_params['usage'].blank?
```

Comment:
What if a user unset the field via purposectl? Would be be prevented from unsetting the role/usage in Katello that way since it's blank?

Reply:
Yep - and that's bad !